control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1128,72 +1142,107 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 }
 
 func (r *HostedControlPlaneReconciler) reconcileEtcd(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage) error {
```

Comment:
Instead of doing the flow control inside this function, can we split out two different etcd paths, like `reconcileManagedEtcd()` and `reconcileUnmanagedEtcd()` and call the right one from the outside to make them more simple/self contained/testable in isolation?

Reply:
done